snapshot-labs/snapshot
# feat: UI render the tenderly simulation button in the tx builder

### Summary


We want tenderly simulations in transaction builder and in the transaction viewer during proposal execution. We also add a feature to warn users if their tx will use gas beyond what our bots will automatically execute.

builder view:
![image](https://github.com/snapshot-labs/snapshot/assets/4429761/f0eab5d6-2549-46fa-a8cb-bd6cf8f2d8bd)
![image](https://github.com/snapshot-labs/snapshot/assets/4429761/17c641c4-08e6-4db6-8ac6-8507d01b528d)

propoosal view:
![image](https://github.com/snapshot-labs/snapshot/assets/4429761/b58c336c-699a-447d-9387-c1f9de3c2e67)
![image](https://github.com/snapshot-labs/snapshot/assets/4429761/5e12f98c-7a4e-4d9e-a6d1-a8e1850ec18d)

gas warning:
![image](https://github.com/snapshot-labs/snapshot/assets/4429761/02a695c5-a7b0-4e67-a4c3-5ca34ca4d242)

private tenderly sim ( should never show, only does in cases where we have misconfigured tenderly):
![image](https://github.com/snapshot-labs/snapshot/assets/4429761/093c54ef-c6df-4cc9-ad7e-2c4a8aa41a7b)

Closes: #

### How to test

1. go to an osnap space, and propose a new transction
2. edit the transaction however you want, see that you can run a simulation
3. propose vote, and see transaction simulator in view

### To-Do
